torch_harmonics/convolution.py

```diff
@@ -167,7 +167,11 @@ def _normalize_convolution_tensor_s2(psi_idx, psi_vals, in_shape, out_shape, ker
                 iidx = torch.argwhere((idx[0] == ik) & (idx[2] == ilat))
                 # normalize, while summing also over the input longitude dimension here as this is not available for the output
                 vnorm = torch.sum(psi_vals[iidx] * q[iidx])
-                psi_vals[iidx] = psi_vals[iidx] / (vnorm + eps)
+                if merge_quadrature:
+                    # the correction factor accounts for the difference in longitudinal grid points when the input vector is upscaled
+                    psi_vals[iidx] = psi_vals[iidx] * q[iidx] * nlon_in / nlon_out / (vnorm + eps)
```

Comment:
Why do we need the nlon_in / nlon_out ?

Reply:
the quadrature weights that are used to normalize the transpose use the output resolution, however quadrature needs to run over the input. This has to be accounted for using this factor.